### ISSUE TITLE
[QDateTime] Month 'forward' button has undesirable behaviour

### ISSUE THREAD

**Issue by dsl101** [quasarframework/quasar#847]
### Software version

Quasar: 0.14
Browsers: Vivaldi 1.11.894.3 (Official Build) (64-bit)

### What did you get as the error?
When a date picker is opened on the 31st of a month which _precedes_ a month with only 30 days (e.g. 2017-08-31), clicking the right-arrow to move to the next month causes a jump by 2 months.

### What were you expecting?
I would expect a more natural behaviour would be to jump to the last day of the next month (i.e. in this case 2017-09-30)

This problem seems to stem from the javascript implementation of `setMonth()`. Try this in a browser console:

```
d = new Date('2017-08-31')
d.getMonth()  // 7
d.setMonth(8)
d.getMonth()  // 9
````

I'm working on a PR for fixing this, but it's not obvious—I'm not sure quite how the `setMonth()` function in `InlineDatetimeMat.vue` works—`this.month` is 1–based, whereas the javascript `Date().setMonth()` is 0–based. Need to look closer and out of time right now.

**Comment by dsl101** [comment]
See PR #850 - I think this solves the problem both forward and backwards.

**Comment by rstoenescu** [comment]
Decided to go with another approach, fixing some "date" utils as well.
Fix is available in latest edge version and future v0.15.
